weldx/core.py

```diff
@@ -346,9 +346,29 @@ def __repr__(self):
             )
         return representation + f"Units:\n\t{self.units}\n"
 
+    @staticmethod
+    def _check_data_array(data_array: xr.DataArray):
+        """Raise an exception if the 'DataArray' can't be used as 'self._data'."""
+        if "time" != data_array.dims[0]:
+            raise ValueError("The first dimension of the DataArray must be 'time'.")
+
+        if "time" not in data_array.coords:
+            raise ValueError("The 'DataArray' does not specify time values.")
+
+        try:
+            ut.to_pandas_time_index(data_array.coords.get("time").data)
+        except TypeError:
+            raise TypeError(
+                "The time values of the 'DataArray' must be convertible to a "
+                "'pandas.TimedeltaIndex'."
+            )
+
+        if not isinstance(data_array.data, pint.Quantity):
+            raise TypeError("The data of the 'DataArray' must be a 'pint.Quantity'.")
+
```

Comment:
I think this should be a place to use `util.xr_check_coords`

Reply:
Not sure about this. We have to create a reference `DataArray` just to replace 3 (the time position case will be removed) simple checks. Feels to be too much overhead to me. Additionally, the error messages are probably less clear to the user.

Reply:
why would we have to create another `DataArray`?
```python
xr_check_coords(data_array, {time={"dtype": ["timedelta64[ns]"]})
```

I prefer this for consistency over individual implementations just for different error messages (maybe throw one meaningful error when `xr_check_coords` fails, that should be enough)
If the error messages turn out not to be helpful enough we can reassess later

Reply:
> why would we have to create another `DataArray`?

I somehow misread the docstring 🤦 My bad...